tests/e2e/create/create_test.go

```diff
@@ -0,0 +1,456 @@
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Test 'create' command", Ordered, func() {
+	Context("Given 'modulectl create' command", func() {
+		var cmd createCmd
+		It("When invoked without any args", func() {
+			cmd = createCmd{}
+		})
+
+		It("Then the command should fail", func() {
+			err := cmd.execute()
+			Expect(err).Should(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("failed to read file module-config.yaml: open module-config.yaml: no such file or directory"))
+		})
+	})
+
+	Context("Given 'modulectl create' command", func() {
+		var cmd createCmd
+		It("When invoked with '--module-config-file' using file with missing name", func() {
+			cmd = createCmd{
+				moduleConfigFile: missingNameConfig,
+			}
+		})
+		It("Then the command should fail", func() {
+			err := cmd.execute()
+			Expect(err).Should(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("invalid Option: opts.ModuleName must not be empty"))
+		})
+	})
+
+	Context("Given 'modulectl create' command", func() {
+		var cmd createCmd
+		It("When invoked with '--module-config-file' using file with missing channel", func() {
+			cmd = createCmd{
+				moduleConfigFile: missingChannelConfig,
+			}
+		})
+		It("Then the command should fail", func() {
+			err := cmd.execute()
+			Expect(err).Should(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("invalid Option: opts.ModuleChannel must not be empty"))
+		})
+	})
+
+	Context("Given 'modulectl create' command", func() {
+		var cmd createCmd
+		It("When invoked with '--module-config-file' using file with missing version", func() {
+			cmd = createCmd{
+				moduleConfigFile: missingVersionConfig,
+			}
+		})
+		It("Then the command should fail", func() {
+			err := cmd.execute()
+			Expect(err).Should(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("invalid Option: opts.ModuleVersion must not be empty"))
+		})
+	})
+
+	Context("Given 'modulectl create' command", func() {
+		var cmd createCmd
+		It("When invoked with '--module-config-file' using file with missing manifest", func() {
+			cmd = createCmd{
+				moduleConfigFile: missingManifestConfig,
+			}
+		})
+		It("Then the command should fail", func() {
+			err := cmd.execute()
+			Expect(err).Should(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("failed to parse module config: failed to value module config: manifest path must not be empty: invalid Option"))
+		})
+	})
+
+	Context("Given 'modulectl create' command", func() {
+		var cmd createCmd
+		It("When invoked with '--module-config-file' using valid file", func() {
+			cmd = createCmd{
+				moduleConfigFile: minimalConfig,
+			}
+		})
+		It("Then the command should succeed", func() {
+			Expect(cmd.execute()).To(Succeed())
+
+			By("And no module template file should be generated")
+			currentDir, err := os.Getwd()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(filesIn(currentDir)).Should(Not(ContainElement("template.yaml")))
+		})
+	})
```

Comment:
Just for my understanding, what is the use case for not writing the ModuleTemplate? 

Also, docs state:

> `Path to write the ModuleTemplate file to, if the module is uploaded to a registry (default "template.yaml"). (default "template.yaml")`

What does the deafult `template.yaml` mean there? If I don't specify the option, nothing is written. If I specify the option, it says I must provide a value

Reply:
The template file will be generated only when `registry` flag is given with default name. With the `output` flag you can further define name and path of the output template file.